packages/process/src/node/raw-process.ts

```diff
@@ -50,30 +54,32 @@ export interface RawProcessFactory {
     (options: RawProcessOptions | RawForkOptions): RawProcess;
 }
 
-/**
- * A Node stream like `/dev/null`.
- *
- * Writing goes to a black hole, reading returns `EOF`.
- */
-class DevNullStream extends stream.Duplex {
-    // tslint:disable-next-line:no-any
-    _write(chunk: any, encoding: string, callback: (err?: Error) => void): void {
-        callback();
-    }
-
-    _read(size: number): void {
-        // tslint:disable-next-line:no-null-keyword
-        this.push(null);
-    }
-}
-
 @injectable()
 export class RawProcess extends Process {
 
-    readonly input: stream.Writable;
-    readonly output: stream.Readable;
-    readonly errorOutput: stream.Readable;
-    readonly process: ChildProcess;
+    /**
+     * @deprecated use `inputStream` instead.
+     */
+    get input(): stream.Writable { return this.inputStream; }
+
+    /**
+     * @deprecated use `outputStream` instead.
+     */
+    get output(): stream.Readable { return this.outputStream; }
+
+    /**
+     * @deprecated use `errorStream` instead.
+     */
+    get errorOutput(): stream.Readable { return this.errorStream; }
+
+    /**
+     * If the process fails to launch, it will be undefined.
+     */
+    readonly process: ChildProcess | undefined;
```

Comment:
it is a breaking change

Reply:
Well, currently the process field can already be undefined, but isn't caught by the type system, so do I revert to the bogus typing?

Reply:
`process` is assigned here:

https://github.com/theia-ide/theia/blob/bff5375ebbef6228dab924f34c05649b1d790cd5/packages/process/src/node/raw-process.ts#L102-L112

But if the `fork` or `spawn` fail, we cannot assign `process` anymore, leaving it undefined:

https://github.com/theia-ide/theia/blob/bff5375ebbef6228dab924f34c05649b1d790cd5/packages/process/src/node/raw-process.ts#L135-L143

Reply:
I wonder why in practice it does not cause any issues. Do you know clients using `process`? If  there are not it would be fine to break.

Reply:
@akosyakov this was the only client in the main sources:

https://github.com/theia-ide/theia/blob/50693c6d380ef3e0aef657550c1f4fd91940df0d/packages/languages/src/node/language-server-contribution.ts#L81

But I think the reason it wasn't failing in practice is because for it to fail, something before has to go wrong, so it just adds one issue on top of another, its like "failing twice". But when things go right this is not an issue.